public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+    let detectorBody = {
+      name: values.name,
+      description: values.description,
+      indices: values.indices,
+      resultIndex: resultIndex,
+      filterQuery: formikToFilterQuery(values),
+      uiMetadata: {
+        features: {
+          ...featuresToUIMetadata(featureListToUIMetadata(values.featureList)),
+        },
+        filters: get(values, 'filters', []),
+      },
+      featureAttributes: formikToFeatureAttributes(values.featureList),
+      timeField: values.timeField,
+      detectionInterval: values.detectionInterval,
+      windowDelay: values.windowDelay,
+      shingleSize: values.shingleSize,
+      categoryField: !isEmpty(values?.categoryField)
+        ? values.categoryField
+        : undefined,
+    } as Detector;
+    // Optionally add detection date range
+    if (detectionDateRange) {
+      detectorBody = {
+        ...detectorBody,
+        //@ts-ignore
+        detectionDateRange: detectionDateRange,
+      };
+    }
+
+    return detectorBody;
+  }
+
+  function featureListToUIMetadata(featureList) {
+    return featureList.map(function (value) {
+      const oneFeature = {
+        featureId: value.featureId,
+        featureName: value.featureName,
+        featureType: value.featureType,
+        featureEnable: true,
+        importance: 1,
+        aggregationBy: value.aggregationBy,
+        aggregationQuery:
+          value.featureType === 'simple_aggs'
+            ? '{}'
+            : formikToUIMetadataAggregation(
+                value.featureId,
+                value.aggregationBy,
+                value.aggregationOf[0].label
+              ),
+        newFeature: true,
+        aggregationOf: formikToAggregationOf(value),
+      };
+
+      return oneFeature;
+    });
+  }
+
+  const initialDetectorValue = {
+    name: detectorNameFromVis,
+    indices: [embeddable.vis.data.aggs.indexPattern.title],
+    timeField: embeddable.vis.data.indexPattern.timeFieldName,
+    detectionInterval: {
+      period: { interval: intervalValue, unit: UNITS.MINUTES },
+    },
+    windowDelay: {
+      period: { interval: delayValue, unit: UNITS.MINUTES },
+    },
+    shingleSize: shingleSizeValue,
+    filterQuery: { match_all: {} },
+    description: '',
+    resultIndex: undefined,
+    filters: [],
+    featureList: featureListToFormik(featureList),
+    categoryFieldEnabled: false,
+    realTime: true,
+    historical: false,
+  };
+
+  const handleAssociate = async (detectorId: string) => {
+    console.log('inside handleAssociate');
+    enum VisLayerTypes {
+      PointInTimeEvents = 'PointInTimeEvents',
+    }
+
+    const fn = {
+      type: VisLayerTypes.PointInTimeEvents,
+      name: 'overlay_anomalies',
+      args: {
+        detectorId: detectorId,
+      },
+    } as VisLayerExpressionFn;
+
+    const savedObjectToCreate = {
+      title: 'test-title',
+      pluginResourceId: detectorId,
+      visId: embeddable.vis.id,
+      visLayerExpressionFn: fn,
+    } as ISavedAugmentVis;
```

Comment:
The latest `ISavedAugmentVis` interface has changed. See [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4006/files#diff-f3a7084285f53c21fad7d46ef7e11a91926e5568f3ac0c9ef9df96f015f99f2a)